.github/workflows/ci_pipeline.yml

```diff
@@ -169,6 +169,58 @@ jobs:
           path: |
             data_libpressio
             coverage_libpressio.dat
+
+  user_firedrake_tests:
+    runs-on: ubuntu-latest
+    container:
+      image: firedrakeproject/firedrake-vanilla:latest
+      options: --user root
+      volumes:
+        - ${{ github.workspace }}:/repositories
+    defaults:
+      run:
+        shell: bash -l {0}
+    steps:
+      - name: Checkout pySDC
+        uses: actions/checkout@v4
+        with: 
+          path: ./pySDC
+      - name: Checkout gusto
```

Comment:
Do you need gusto in this PR?

Reply:
We don't need this here, but the pipeline was a bit of a mess to setup because I am checking out two repositories and then installing them. I thought there is not much of a point in untangling this and then reverting this for the follow-up Gusto PR. Unless you would want two separate test environments for Gusto and Firedrake? But I can remove Gusto here anyways if you want.

Reply:
Nope, all good.